packages/frontend/components/scaffold-eth/Contract/InputUI.tsx

```diff
@@ -0,0 +1,102 @@
+import { ethers } from "ethers";
+import { FunctionFragment } from "ethers/lib/utils";
+import React, { Dispatch, FC, ReactElement, SetStateAction } from "react";
+import Address from "../Address";
+
+type ParamType = {
+  name: string | null;
+  type: string;
+};
+
+interface IInputUI {
+  setForm: Dispatch<SetStateAction<Record<string, any>>>;
+  form: Record<string, any>;
+  stateObjectKey: string;
+  paramType: ParamType;
+  functionFragment: FunctionFragment;
+}
+
+// TODO use AddressInput component instead
+const InputUI: FC<IInputUI> = ({ setForm, form, stateObjectKey, paramType }) => {
+  let buttons: ReactElement = <></>;
+
+  // TODO maybe abstract each `div` as a small util Component, use switch case and cleanUp
```

Comment:
Should we abstract each `div`'s inside the if condition and maybe be put it in a `utilsComponents` file to make it cleaner?

Reply:
Yeah, I like the idea!

We could do a switch (instead of all if-elses):

```tsx
  switch (paramType.type) {
    case "bytes32":
      // call the util
      break;
  }
```

Regarding the utils... maybe we can have:

- **utilsContract** (currently named "utils")
- **utilsDisplay** (currently named "displayUtils")
- **utilsInputs** / **utilsComponents**... or whatever you like (depending if we only use it for the getting the "buttons" or algo the "inputs)